src/hooks/useWithdrawLiquidityHandler.tsx

```diff
@@ -1,372 +1,263 @@
+      setsliderPercentage(percentage)
+    } catch (err: any) {
+      notifyError(err.toString())
+    }
   }
 
   async function sliderPercentChange(value: number | number[]) {
-    if (typeof value === 'number') {
-      setsliderPercentage(value)
+    try {
+      if (typeof value === 'number') {
+        setEnableApprove(value > 0 && parseInt(totalLiquidityAmt) > 0)
+        setsliderPercentage(value)
+      }
+    } catch (err: any) {
+      notifyError(err.toString())
     }
   }
 
-  async function approveTransaction() {
-    // Position Manager contract address
-    let address =
-      process.env.NEXT_PUBLIC_ENVIRONMENT === 'local'
-        ? process.env.NEXT_PUBLIC_POSITION_MANAGER_ADDRESS || ''
-        : DEPOSIT_POOL_ADDRESS
+  useEffect(() => {
     if (provider) {
-      if (process.env.NEXT_PUBLIC_ENVIRONMENT === 'local') {
-        if (accountInfo?.address) {
-          setPosManager(new ethers.Contract(address, PosManager.abi, provider.getSigner(accountInfo.address)))
-        } else {
-          setPosManager(new ethers.Contract(address, PosManager.abi, provider))
-        }
-      } else {
-        // if (accountInfo?.address) {
-        //   setDepPool(new ethers.Contract(address, DepPool.abi, provider.getSigner(accountInfo.address)))
-        // } else {
-        //   setDepPool(new ethers.Contract(address, DepPool.abi, provider))
-        // }
+      setInvalidBtnText('Confirm')
+      if (!positionManager && POSITION_MANAGER_ADDRESS) {
+        setPositionManager(
+          new ethers.Contract(
+            POSITION_MANAGER_ADDRESS,
+            PositionManagerJSON.abi,
+            accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+          )
+        )
       }
-    } else {
-      notifyError('Please connect wallet')
-    }
 
-    if (!accountInfo || !accountInfo.address) {
-      notifyError('Wallet not connected.')
-      return
-    }
-    if (accountInfo && accountInfo?.address) {
-      if (posManager === null) {
-        return
-      }
-      approveWithdraw(posManager, posManager.address)
-        .then(() => {
-          setEnableRemove(true)
-        })
-        .catch((err: any) => {
-          setEnableRemove(false)
-          notifyError(err)
-        })
-    } else {
-      if (depPool === null) {
-        return
+      if (!gammaPoolContract && poolAddress) {
+        setGammaPoolContract(
+          new ethers.Contract(
+            poolAddress.toString(),
+            GammaPool.abi,
+            accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+          )
+        )
       }
-      approveWithdraw(depPool, depPool.address)
-        .then(() => {
-          setEnableRemove(true)
-        })
-        .catch((err: any) => {
-          setEnableRemove(false)
-          notifyError(err)
-        })
     }
+  }, [provider])
+
+  async function approveTransaction() {
```

Comment:
can't we use validation.tsx approve function? GammaPool should be ERC20

Reply:
Done